### PR TITLE
https: Unhide alt svc cache

### PR DESCRIPTION
**Pull request by RyanTheOptimist**
http: Unhide the AlternateProtocolCache configuration
and move some docs from http3_upstream.md to connection_pooling.rst

Risk Level: N/A - Docs only
Testing: N/A - Docs only
Docs Changes: Unhide the AlternateProtocolCache configuration and move some docs from http3_upstream.md to connection_pooling.rst
Release Notes: Added
Platform Specific Features: N/A

### CODE REVIEW

**Comment by RyanTheOptimist** [comment]
I'm not quite sure how much of the implementation details should be moved to the .rst file so I decided to move the config, but keep the impl details in the .md. Let me know if you'd prefer a different approach.

File: source/docs/http3_upstream.md [hunk 1]
```diff
@@ -37,24 +37,3 @@ alternate protocols cache, stream requets will be sent directly to the mixed con
 If, however, HTTP/3 is advertised and working then the request will be sent to the HTTP/3
 connection pool. After 300ms, if this request has not succeeded, then the request will also be
 sent to the mixed pool, and whichever succeeds first will be passed back up to the caller.
-
-### Required configuration
```
Comment (alyssawilk) [inline source/docs/http3_upstream.md:41]: Optionally, I think it's worth leaving this here but your call.

**Review by alyssawilk — COMMENTED** [review]
Nice!  thanks for the follow-up

**Review by RyanTheOptimist — COMMENTED** [review]
Thanks!

File: docs/root/intro/arch_overview/upstream/connection_pooling.rst [hunk 2]
```diff
@@ -69,10 +68,21 @@ then 300ms later, if a QUIC connection is not established, will also attempt to
...
+If an alternate protocol cache is configured via the
+:ref:`alternate_protocols_cache_options <envoy_v3_api_msg_extensions.upstreams.http.v3.AlternateProtocolsCacheOptions>`
+then HTTP/3 connections will only be attempted to servers which
+advertise HTTP/3 support either via `HTTP Alternative Services <https://tools.ietf.org/html/rfc7838>`, (eventually
+the `HTTPS DNS resource record<https://datatracker.ietf.org/doc/html/draft-ietf-dnsop-svcb-https-04>` or "QUIC hints"
+which will be manually configured).
+If no such advertisement exists, then HTTP/2 or HTTP/1 will be used instead.
+
+If no alternate protocol cache is configured, then HTTP/3 connections will be attempted to
+all servers, even though which do not advertise HTTP/3.
```
Comment (alyssawilk) [inline docs/root/intro/arch_overview/upstream/connection_pooling.rst:80]: though -> those?

Comment (RyanTheOptimist) [reply docs/root/intro/arch_overview/upstream/connection_pooling.rst]: *facepalm* Done.

**Review by alyssawilk — COMMENTED** [review]
alas, docs build is now unhappy?

**Comment by RyanTheOptimist** [comment]
> alas, docs build is now unhappy?

Ok, finally got the ref targets sorted out. Should be good now.

**Review by alyssawilk — APPROVED** [review]

**Comment by alyssawilk** [comment]
Ah, looks like Mark is out.  @adisuissa would you be willing to API stamp this one?

**Review by adisuissa — APPROVED** [review]
/lgtm api

**Pull request merged**